src/porepy/models/protocol.py

```diff
@@ -835,3 +888,85 @@ class PorePyModel(
             runtime, since it is not an abstract base class.
 
         """
+
+    class CompositionalFlowModelProtocol(PorePyModel, Protocol):
```

Comment:
I was less involved towards the end of the work on the protocols. I don't immediately see strong reasons to avoid this, but wonder whether it could be better to keep them separate, i.e., not inherit from PorePyModel. Also, let's consult @Yuriyzabegaev

Reply:
I am not opposed to introducing this. @Yuriyzabegaev: Do you have strong opinions?

Reply:
Do the mentioned concerns hold despite splitting the pseudo-protocol and the actual protocol using `TYPE_CHECKING`?
The one used at runtime does not inherit anything (as per intended use), the one used for type checking (actual protocol) does inherit from `PorePyModel` (again- the actual protocol).